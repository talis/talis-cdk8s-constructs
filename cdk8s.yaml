language: typescript
imports:
  - k8s@1.30.0
