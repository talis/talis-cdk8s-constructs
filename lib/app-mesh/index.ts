export * from "./ingress-gateway-nlb";
export * from "./app-service";
