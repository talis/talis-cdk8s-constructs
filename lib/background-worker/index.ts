export * from "./background-worker-props";
export * from "./background-worker";
