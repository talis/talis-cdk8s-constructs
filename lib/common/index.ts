export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
