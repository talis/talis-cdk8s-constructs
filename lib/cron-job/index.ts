export * from "./cron-job-props";
export * from "./cron-job";
