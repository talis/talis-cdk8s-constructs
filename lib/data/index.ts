export * from "./config-map";
export * from "./secret";
export * from "./factory";
