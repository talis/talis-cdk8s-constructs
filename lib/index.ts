export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
