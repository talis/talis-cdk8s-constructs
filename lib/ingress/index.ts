export * from "./ingress-props";
export * from "./ingress";
