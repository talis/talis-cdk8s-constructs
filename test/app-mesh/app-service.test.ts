import { Chart, Testing } from "cdk8s";
import { AppService, AppServiceProps } from "../../lib";

const requiredProps: AppServiceProps = {
  app: "myapp",
  port: 5000,
  instance: "server",
  hostname: "myapp.example.com",
  virtualNodeTargets: [
    {
      name: "myapp",
      namespace: "default",
      weight: 100,
    },
  ],
  gatewayRouteLabels: {
    "aws.tfs.engineering/appMeshGateway": "12345",
  },
};

function synthAppService(
  props: AppServiceProps,
  chart: Chart = Testing.chart(),
) {
  new AppService(chart, "app-service-test", props);
  return Testing.synth(chart);
}

describe("AppService", () => {
  describe("Props", () => {
    test("Required props", () => {
      const results = synthAppService(requiredProps);
      expect(results).toMatchSnapshot();
    });
  });

  describe("Props Validation", () => {
    test("throws error when weight sum is greater than 100", () => {
      const props: AppServiceProps = {
        ...requiredProps,
        virtualNodeTargets: [
          {
            name: "myapp",
            namespace: "default",
            weight: 80,
          },
          {
            name: "myapp",
            namespace: "newnamespace",
            weight: 40,
          },
        ],
      };
      expect(() => synthAppService(props)).toThrowError();
    });
  });
});
