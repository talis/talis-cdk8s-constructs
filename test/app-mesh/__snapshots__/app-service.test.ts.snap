// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppService > Props > Required props 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "app": "myapp",
        "instance": "server",
      },
      "name": "myapp",
    },
    "spec": {
      "ports": [
        {
          "name": "http",
          "port": 5000,
        },
      ],
    },
  },
  {
    "apiVersion": "appmesh.k8s.aws/v1beta2",
    "kind": "VirtualRouter",
    "metadata": {
      "labels": {
        "app": "myapp",
        "instance": "server",
      },
      "name": "myapp-router",
    },
    "spec": {
      "listeners": [
        {
          "portMapping": {
            "port": 5000,
            "protocol": "http",
          },
        },
      ],
      "routes": [
        {
          "httpRoute": {
            "action": {
              "weightedTargets": [
                {
                  "virtualNodeRef": {
                    "name": "myapp",
                    "namespace": "default",
                  },
                  "weight": 100,
                },
              ],
            },
            "match": {
              "prefix": "/",
            },
          },
          "name": "myapp-route",
        },
      ],
    },
  },
  {
    "apiVersion": "appmesh.k8s.aws/v1beta2",
    "kind": "VirtualService",
    "metadata": {
      "labels": {
        "app": "myapp",
        "instance": "server",
      },
      "name": "myapp-virtual-service",
    },
    "spec": {
      "awsName": "myapp.undefined.svc.cluster.local",
      "provider": {
        "virtualRouter": {
          "virtualRouterRef": {
            "name": "myapp-router",
          },
        },
      },
    },
  },
  {
    "apiVersion": "appmesh.k8s.aws/v1beta2",
    "kind": "GatewayRoute",
    "metadata": {
      "labels": {
        "app": "myapp",
        "aws.tfs.engineering/appMeshGateway": "12345",
        "instance": "server",
      },
      "name": "myapp-gateway-route",
    },
    "spec": {
      "httpRoute": {
        "action": {
          "target": {
            "virtualService": {
              "virtualServiceRef": {
                "name": "myapp-virtual-service",
              },
            },
          },
        },
        "match": {
          "hostname": {
            "exact": "myapp.example.com",
          },
        },
      },
    },
  },
]
`;
